core/http/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/transaction/Transaction.java

```diff
@@ -113,13 +117,12 @@ UUID getID() {
 	void begin(IsolationLevel level)
 		throws InterruptedException, ExecutionException
 	{
-		Future<Boolean> result;
-		synchronized (futures) {
-			result = executor.submit(() -> {
-				txnConnection.begin(level);
-				return true;
-			});
+		Future<Boolean> result = executor.submit(() -> {
+			txnConnection.begin(level);
+			return true;
+		});
 
+		synchronized (futures) {
```

Comment:
I moved all of the executor.submit commands outside of synchronized(futures) as the order of submit isn't guaranteed to be the order that the executor will actually pick them in, and the JVM gives no guarantees about the order of releasing/locking monitors for synchronized blocks. I can revert this if it was intentionally done for another reason.